pkg/asset/installconfig/azure/validation.go

```diff
@@ -188,11 +191,28 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 		if ultraSSDCapability == "" {
 			ultraSSDCapability = defaultUltraSSDCapability
 		}
-		if vmNetworkingType == "" {
-			vmNetworkingType = defaultVMNetworkingType
+
+		capabilities, err = client.GetVMCapabilities(context.TODO(), instanceType, ic.Azure.Region)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, err.Error()))
+		}
+
+		if vmNetworkingType == "" && capabilities != nil {
+			val, ok := capabilities[string(aztypes.AcceleratedNetworkingEnabled)]
+			if ok {
+				if strings.EqualFold(val, "True") {
+					vmNetworkingType = string(aztypes.VMnetworkingTypeAccelerated)
+				} else {
+					vmNetworkingType = string(aztypes.VMNetworkingTypeBasic)
+				}
+			}
+
```

Comment:
This part can be removed, can't it? You're assigning a value to `vmNetworkingType` and then validating the value is correct (which it should be, since you assigned the correct value). Later on, when provisioning the machines, you're going to do the same assignment according to the capabilities again.

Reply:
Sounds good to me. Will skip validation of the vmNetworkignType when set to `""`.